### PR TITLE
Fix #2989: Add delete apis for tests

### LINKED ISSUE #2989: Add delete API for tests
**Feature**
Add an option to delete the tests
1. Return one result when its added instead of all the list
2. If the same timestamp sent for a result we should update it instead of adding 
3. Possibly generate change event on a result addition

**Describe the task**

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
See #2989 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by harshach** [comment]
Test failure not related to the code in this PR

File: catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "updatedAt": {
+      "description": "Last update time corresponding to the new version of the entity in Unix epoch time milliseconds.",
+      "$ref": "../../type/basic.json#/definitions/timestamp"
+    },
+    "updatedBy": {
+      "description": "User who made the update.",
+      "type": "string"
+    }
+  },
+  "required": ["testCase", "tableName"],
```
Comment (pmbrull) [inline catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json:39]: As we are running the PUT to `tables/{id}/tableTest`, do we need to pass the `tableName` as a param?

File: catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "properties": {
+    "description": {
+      "description": "Description of the testcase.",
+      "type": "string"
+    },
+    "tableName": {
+      "description": "Table Name for which this test applies.",
+      "type": "string"
+    },
+    "tableTestCase": {
```
Comment (pmbrull) [inline catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json:16]: I'd either rename this to `testCase` (as we currently have with `ColumnTest`) or rename Column's `testCase` to `columnTestCase` to follow the same naming convention

**Comment by pmbrull** [comment]
Hi! As we discussed, some points that we could use this PR as well to fix:

If possible, we should update as well:
- From `columnValuesToMatchRegex.json`: the title should be fixed to match the file name
- From `basic.json`: Update `status` property name to `testCaseStatus`
- From `tableTest.json`: `tableTestCase` property renamed to `testCase` to match the same definition in column
- Review `title` for the implemented table tests

I'll update this list if I find anything else. Thanks 🙏

File: catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json (hunk 1 above)
Comment (pmbrull) [reply catalog-rest-service/src/main/resources/json/schema/api/tests/createTableTest.json:39]: @harshach I believe we can remove this `tableName` from the create API as well

**Review by pmbrull — APPROVED** [review]
Thanks for the PR!! I think there's just one pending comment, but other than that looks great 🚀

**Pull request merged**